x-pack/plugins/observability_solution/infra/public/components/asset_details/tabs/dashboards/actions/save_dashboard_modal.tsx

```diff
@@ -0,0 +1,263 @@
+            });
+
+            if (result && !isUpdateLoading) {
+              notifications.toasts.success(getEditSuccessToastLabels(newDashboard.label));
+            }
+          } else {
+            const result = await createCustomDashboard({
+              assetType,
+              dashboardSavedObjectId: newDashboard.value,
+              dashboardFilterAssetIdEnabled: assetNameEnabled,
+            });
+
+            if (result && !isCreateLoading) {
+              notifications.toasts.success(getLinkSuccessToastLabels(newDashboard.label));
+            }
+          }
+
+          setUrlState({ dashboardId: newDashboard.value });
+          onRefresh();
+        }
+      } catch (error) {
+        notifications.toasts.danger({
+          title: i18n.translate('xpack.infra.customDashboards.addFailure.toast.title', {
+            defaultMessage: 'Error while adding "{dashboardName}" dashboard',
+            values: { dashboardName: newDashboard.label },
+          }),
+          body: error.message,
+        });
+      }
+      onClose();
+    },
+    [
+      selectedDashboard,
+      onClose,
+      isEditMode,
+      setUrlState,
+      onRefresh,
+      updateCustomDashboard,
+      assetType,
+      currentDashboard?.id,
+      assetNameEnabled,
+      isUpdateLoading,
+      notifications.toasts,
+      createCustomDashboard,
+      isCreateLoading,
+    ]
+  );
+
+  return (
+    <EuiModal onClose={onClose} data-test-subj="infraSelectCustomDashboard">
+      <EuiModalHeader>
+        <EuiModalHeaderTitle>
+          {isEditMode
+            ? i18n.translate('xpack.infra.customDashboards.selectDashboard.modalTitle.edit', {
+                defaultMessage: 'Edit dashboard',
+              })
+            : i18n.translate('xpack.infra.customDashboards.selectDashboard.modalTitle.link', {
+                defaultMessage: 'Select dashboard',
+              })}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+
+      <EuiModalBody>
+        <EuiFlexGroup direction="column" justifyContent="center">
+          <EuiComboBox
+            isLoading={status === FETCH_STATUS.LOADING}
+            isDisabled={status === FETCH_STATUS.LOADING || isEditMode}
+            placeholder={i18n.translate(
+              'xpack.infra.customDashboards.selectDashboard.placeholder',
+              {
+                defaultMessage: 'Select dashboard',
+              }
+            )}
+            singleSelection={{ asPlainText: true }}
+            options={options}
+            selectedOptions={selectedDashboard}
+            onChange={(newSelection) => setSelectedDashboard(newSelection)}
+            isClearable={true}
+          />
+
+          <EuiSwitch
+            css={{ alignItems: 'center' }}
+            compressed
+            label={
+              <p>
+                {i18n.translate('xpack.infra.customDashboard.addDashboard.useContextFilterLabel', {
+                  defaultMessage: 'Filter by host name',
+                })}{' '}
```

Comment:
nit:
```suggestion
                })}
```

Reply:
This adds space after the filter text.
Without it, the icon will be too close:
<img width="396" alt="image" src="https://github.com/elastic/kibana/assets/14139027/3e241d6b-abf4-4edb-9b92-92bb20bfde27">

Reply:
You could use `margin` instead, then.